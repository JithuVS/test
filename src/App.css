.App {
  text-align: center;
  margin-top: 20px;
}

.Container{
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 10px;
  grid-auto-rows: 20rem;
}

.item{
  border: 2px red solid;
}

.fields {
  width: 40%;
  padding: 12px;
  margin: 10px;
  border: 1px solid #ccc;
  border-radius: 6px;
  box-sizing: border-box;
}

.left{
  text-align: left;
}

.block {
  display: 'block';
}

.button {
  padding: 10px;
  margin: 5px;
}

.margin {
  margin: 2rem;
}