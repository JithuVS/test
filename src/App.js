import { Provider } from "react-redux";
import "./App.css";
import Axios from "./components/Axios/Axios";
import Counter from "./components/Counter/Counter";
import Post from "./components/Post/Post";
import { store } from "./components/Post/Store";
import Reducer from "./components/Reducer/Reducer";
import Dom from "./components/Dom/Dom";
import Misc from "./components/Misc/Misc";


function App() {
  return (
    <div className="App">
      <div className="Container">
        
      <div className="item">
          <h3 id='dom'>Dom/Props/Statee</h3>
          <Dom learn="Jest" />
        </div>
        <div className="item">
          <h3 data-test='counter'>Counter/Simulate</h3>
          <Counter />
        </div>
        <div className="item">
          <h3>Reducer</h3>
          <Reducer />
        </div>
        <div className="item">
          <h3>Redux</h3>
          <Provider store={store}>
            <Post />
          </Provider>
        </div>
        <div className="item">
          <h3>Async/Axios/Mock</h3>
          <Axios />
        </div>
        <div className="item">
          <h3>Misc</h3>
          <Misc/>
        </div>
      </div>
    </div>
  );
}

export default App;
