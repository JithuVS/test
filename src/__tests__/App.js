// import { render, screen } from '@testing-library/react';
import Enzyme, { mount, shallow, render} from 'enzyme';
import App from '../App';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import jsonTo from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

test('renders learn react link', () => {
  const wrapper = mount(<App />);
  // console.log(wrapper.debug());

  const  dom = wrapper.find('.margin');
  expect(dom.text()).toBe('Learn');
});

test('renders counter', () => {
  const wrapper = mount(<App />);
  // console.log(wrapper.debug());

  const  dom = wrapper.find('[data-test="counter"]');
  expect(dom.text()).toBe('Counter/Simulate');
});


test('renders learn react link', () => {
  const wrapper = shallow(<App />);
  // console.log(wrapper.debug());
});



it.skip('snapshot',()=>{
  const wrapper = mount(<App/>);
  expect(jsonTo(wrapper)).toMatchSnapshot();
})


