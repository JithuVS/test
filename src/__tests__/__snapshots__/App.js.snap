// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <div
      className="Container"
    >
      <div
        className="item"
      >
        <h3
          id="dom"
        >
          Dom/Props/Statee
        </h3>
        <Dom
          learn="Jest"
        >
          <section>
            <div
              className="margin"
            >
              Learn
            </div>
            <div
              data-test="data"
            >
              <span
                key="0.4954156433184509"
              >
                Jest
                 
              </span>
            </div>
            <br />
            <br />
            <div
              onClick={[Function]}
            >
              <button
                className="button"
                data-test="css"
              >
                CSS
              </button>
              <button
                className="button"
              >
                HTML
              </button>
              <button
                className="button"
              >
                React
              </button>
              <button
                className="button"
              >
                Redux
              </button>
            </div>
          </section>
        </Dom>
      </div>
      <div
        className="item"
      >
        <h3
          data-test="counter"
        >
          Counter/Simulate
        </h3>
        <Counter>
          <div
            className="app"
          >
            <div>
              <div>
                <h1
                  data-test="count"
                >
                  0
                </h1>
              </div>
              <div>
                <button
                  data-test="decrement"
                  onClick={[Function]}
                  title="-"
                >
                  -
                </button>
                <button
                  data-test="increment"
                  onClick={[Function]}
                  title="+"
                >
                  +
                </button>
              </div>
            </div>
          </div>
        </Counter>
      </div>
      <div
        className="item"
      >
        <h3>
          Reducer
        </h3>
        <Reducer>
          <div>
            <section>
              <form
                className="block"
              >
                <input
                  className="fields"
                  data-test="post-title"
                  id="title"
                  placeholder="Title"
                  type="text"
                />
                 
                <br />
                <br />
                <textarea
                  className="fields"
                  data-test="post-comment"
                  id="comment"
                  placeholder="Comment"
                />
                <br />
                <button
                  data-test="post-button"
                  onClick={[Function]}
                >
                  submit
                </button>
              </form>
              <hr />
              <section>
                <div
                  data-test="list-first"
                  id="1"
                  key="1"
                  onClick={[Function]}
                >
                  first
                  comment
                </div>
              </section>
            </section>
          </div>
        </Reducer>
      </div>
      <div
        className="item"
      >
        <h3>
          Redux
        </h3>
        <Provider
          store={
            Object {
              "@@observable": [Function],
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
        >
          <Post>
            <div>
              <section>
                <form
                  className="block"
                >
                  <input
                    className="fields"
                    data-test="post-title"
                    id="title"
                    placeholder="Title"
                    type="text"
                  />
                   
                  <br />
                  <br />
                  <textarea
                    className="fields"
                    data-test="post-comment"
                    id="comment"
                    placeholder="Comment"
                  />
                  <br />
                  <button
                    data-test="post-button"
                    onClick={[Function]}
                  >
                    submit
                  </button>
                </form>
                <hr />
                <section>
                  <div
                    data-test="list-first"
                    id="1"
                    key="1"
                    onClick={[Function]}
                  >
                    first
                    comment
                  </div>
                </section>
              </section>
            </div>
          </Post>
        </Provider>
      </div>
      <div
        className="item"
      >
        <h3>
          Async/Axios/Mock
        </h3>
        <Axios>
          <section>
            <input
              className="fields"
              id="username"
              type="text"
            />
            <button
              onClick={[Function]}
            >
              Submit
            </button>
            <hr />
            <ul
              className="left"
            />
          </section>
        </Axios>
      </div>
      <div
        className="item"
      >
        <h3>
          Misc
        </h3>
        <Misc>
          <div
            className="left"
          >
            <ul>
              <li>
                Debug: "test:debug": "react-scripts --inspect-brk test --runInBand --no-cache"
              </li>
              <br />
              <li>
                Hooks: jest.fn(), jest-react-hooks-shallow
              </li>
              <br />
              <li>
                extension: test, spec, __tests__
              </li>
              <br />
              <li>
                E2E jest-puppeteer
              </li>
              <br />
            </ul>
          </div>
        </Misc>
      </div>
    </div>
  </div>
</App>
`;
